Sources/Apodini/Information/Information.swift

```diff
@@ -7,36 +7,54 @@
 
 import Foundation
 
+/// Describes a type erased version of an `Information` instance.
+public protocol AnyInformation {
+    /// Internal method turning an `Information` instance into a `AnyInformationEntry`.
+    func entry() -> AnyInformationEntry
+}
+
 
 /// Information describes additional metadata that can be attached to a `Response` or can be found in the `ConnectionContext` in the `@Environment` of a `Handler`.
 @dynamicMemberLookup
-public protocol Information: RawRepresentable where RawValue == String {
-    /// The value associated with the type implementing `Information`
+public protocol Information: AnyInformation {
+    /// The `InformationKey` type uniquely identified this `Information` instance.
+    associatedtype Key: InformationKey
+    /// The value associated with the type implementing `Information`.
     associatedtype Value
     
-    
-    /// A key identifying the type implementing `Information`
-    static var key: String { get }
+
+    /// A key identifying the type implementing `Information`.
+    static var key: Self.Key { get }
     
     
     /// The value associated with the type implementing `Information`
     var value: Value { get }
     
     
-    /// Creeate an `Information` based on a value
-    /// - Parameters:
-    ///   - value: The `Information` value
-    init(_ value: Value)
-    
-    
-    /// Enables developeers to direcly access properties of the `Value` using the `Information`
+    /// Enables developers to directly access properties of the `Value` using the `Information`
     subscript<T>(dynamicMember keyPath: KeyPath<Value, T>) -> T { get }
 }
 
-
-extension Information {
-    /// Enables developeers to direcly access properties of the `Value` using the `Information`
-    public subscript<T>(dynamicMember keyPath: KeyPath<Value, T>) -> T {
+// MARK: dynamicMemberLookup
+public extension Information {
+    /// Enables developers to directly access properties of the `Value` using the `Information`
+    subscript<T>(dynamicMember keyPath: KeyPath<Value, T>) -> T {
         value[keyPath: keyPath]
     }
 }
+
+// MARK: Default Key
+public extension Information {
+    /// By default, the `ObjectIdentifier` of the implementing `Information` type will be used.
+    static var key: ObjectIdentifier {
+        ObjectIdentifier(Self.self)
```

Comment:
That is exactly the point I flagged as "Up to discuss" in my `Addressing the Information API` section of the PR description.

With this PR, we allow for `Information` instances which might not be mapped directly to the corresponding wire format (simply stemming from the fact that we don't know the format of the wire format). Therefore an Exporter will need to explicitly support every `Information` type.

Giving some examples:
* HTTP based exporters provide `AnyHTTPInformation` instances, which HTTP based exporters will obviously map to Headers and response (and vice versa in requests). However other exporters potentially can't make that 1:1 mapping (simply as HTTP Headers are a protocol feature). One could image a WebSocket exporter to map `AnyHTTPInformation` to some json field. For Exporters like an GraphQL or gRPC exporter these informations might not make sense at all(?).  
Therefore this PR makes it clear, that **not** mapping a certain Information is perfectly okay. Handler code should/has to/can be written in a way that accommodates for cases where a certain Information is missing (simply as the subscript returns an Optional). So this direction probably makes sense. The reverse direction might be the one which can be a little bit unintuitive, as there is no guarantee that an Information is mapped into the Response **for all** exporters.  
But that was the case already, in the current version the REST exporter is the only one which maps `Information` instances into the response, all others ignore them. This PR just tries to make that clear a priori.
* Another example, which directly addresses your comment of why `ObjectIdentifier` makes sense even though there is no direct way of mapping it to the respective wire protocol, are `Information` instances which are the result of some translation/mapping from or to the respective wire protocol.
My use case with `ApodiniAuthorization` would be a dedicated `ApodiniAuthorizationInformation` (has nothing to do with the http based `Authorization` information), which can be used to map any of the support authentication types (e.g. a token based or credential based auth) to a generic Information instance which is processable by `ApodiniAuthorization`. Analogous the `ApodiniAuthorization` framework would inject a corresponding Information instance into the response, which the exporter needs to map to the according wire format (the response might be used to transport some multi step authorization information or to annotate e.g. an error with information about the failed authorization).
Again there is the argument, that the Exporter is required to manually add support for this `Information` type (if the respective wire format has in-built authorization mechanism, like http does).

Reply:
@theMomax Maybe to reflect on the second point presented here. Could the same approach be realized using the `Environment`, e.g. using the `@EnvironmentObject` or some keyed environment entry? I assume it would be possible to pass the generic `Authorization` object down the handler evaluation using the environment, but the environment cannot be used to "return" such an environment object back to the Exporter right?

Though rethinking my thought (lol), would @EnvironmentObject be able to gracefully handle cases where the according object isn't present (e.g how the subscript always returns the `Optional<Value>` and the respective handler needs to properly handle cases where the Information is not present).

Reply:
Just to mention the result from today's discussion here. The most desirable approach would be to have only basic information on the InformationSet (e.g. a token, properties required for http basic auth). A stack of delegating handlers would then one-by-one access this basic information and transform it (in steps to be defined by you) to something like an authenticated user object. Each delegating handler can either use `@Environment` or `@EncvironmentObject`s to deliver the generated information to the upper `Handler`s.

The nature of the Information-keys is not really resolved. @PSchmiedmayer would vote for just using `String`. I would propose a system where each exporter defines a public protocol of Keys it can handle. E.g. WebSocket would accept any key that can provide a `String` representation. All the HTTP based exporters would probably want a `ValidHTTPHeaderStringProvidingKey` protocol or similar. When defining an `Information`, the creator of this `Information` should conform it to the "well known" key-protocols that are used by exporters where it actually makes sense to expose this type of information.

Reply:
@PSchmiedmayer I would agree with @theMomax that an approach that is based on InformationKey "classes" would solve the above described issues but would be generic enough at the same time.
A key might be mapped to string or something else. In any case through the introduction of those classes we can easily achieve namespace separation even when the resulting value type (e.g. String) is the same.

The key of a concrete Information instance would then declare conformance to one or multiple key classes, making it available in the according exporters. After all the actual Information implementation is Exporter specific (or at least middleware specific) anyways.

Information basically maps certain metadata of a dedicated protocol into Apodini. Therefore any type of Wire format would declare a InformationKey class if applicable (e.g. the VaporSupport package would declare one for e.g. HTTP Headers).

Every Exporter would then provide the according Information Instances for the applicable protocol Metadata (e.g. HTTP Headers) and would consequentially also only include Information which specifically was declared valid in the response.

Reply:
So, the latest commit implements the concept as envisioned by @theMomax and me.